Basic-Car-Maintenance/Shared/Odometer/ViewModels/OdometerViewModel.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  OdometerViewModel.swift
+//  Basic-Car-Maintenance
+//
+//  Created by Nate Schaffner on 10/15/23.
+//
+
+import Foundation
+import FirebaseFirestore
+import FirebaseFirestoreSwift
+
+@Observable
+class OdometerViewModel {
+    
+    let authenticationViewModel: AuthenticationViewModel
+    
+    var readings = [OdometerReading]()
+    var showAddErrorAlert = false
+    var showErrorAlert = false
+    var isShowingAddOdometerReading = false
+    var isShowingAddView = false
+    var errorMessage: String = ""
+    var vehicles = [Vehicle]()
+
+    init(authenticationViewModel: AuthenticationViewModel) {
+        self.authenticationViewModel = authenticationViewModel
+    }
+    
+    func addReading(_ odometerReading: OdometerReading) {
```

Comment:
I changed how this function works because when you throw the error it doesn't show and dismissed the view, now it properly shows the alert.

You can test this by commenting out the Firestore stuff and add `throw URLError(.badURL)` which would make this always throw, and confirm that the alert is shown